### PR TITLE
Add debug panel

### LINKED ISSUE #390: [Onboarding] Create debug flag
- Create a debug flag mechanism (store flags on localstorage?)
- Optional: add debug GUI with switches
- Add `onboarding` debug flag

### LINKED ISSUE #391: [Onboarding] Create Securitize mock
We need to create a mocked control to bypass the Securitize step in onboarding, when we are in test net.

Use statuses as provided by @offerijns :

```javascript
const addressStatusPermutations: { [key: string]: AddressStatus } = {
  'Needs to KYC': {
    kyc: {
      status: 'none',
      url: 'https://www.google.com/',
    },
    agreements: [],
    linkedAddresses: [],
  },
  'Verified, accreditation needed': {
    kyc: {
      status: 'verified',
      isUsaTaxResident: true,
      accredited: false,
    },
    agreements: [],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Verified, awaiting signature': {
    kyc: {
      status: 'verified',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: false,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Voided agreement': {
    kyc: {
      status: 'verified',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: false,
        counterSigned: false,
        declined: false,
        voided: true,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Declined agreement': {
    kyc: {
      status: 'verified',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: false,
        counterSigned: false,
        declined: true,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Counter-signed agreement, awaiting KYC': {
    kyc: {
      status: 'processing',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: true,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Counter-signed agreement, KYC requires updates': {
    kyc: {
      status: 'updates-required',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: true,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Counter-signed agreement, KYC rejected': {
    kyc: {
      status: 'rejected',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: true,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Counter-signed agreement, KYC expired': {
    kyc: {
      status: 'expired',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: true,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: [],
  },
  'Has >1 addresses': {
    kyc: {
      status: 'verified',
    },
    agreements: [
      {
        name: 'DROP Subscription Agreement',
        tranche: 'senior',
        provider: 'docusign',
        providerTemplateId: 'abc',
        signed: true,
        counterSigned: false,
        declined: false,
        voided: false,
      },
    ],
    showNonSolicitationNotice: true,
    linkedAddresses: ['0x0A735602a357802f553113F5831FE2fbf2F0E2e0'],
  },
}
```

### PR DESCRIPTION
**Pull request by onnovisser**
image

Adds a debug panel to replace individual debug flags passed through the URL. Enabled by adding `?debug` onto the URL,
closes #390 
Also adds mock onboarding data, closes #391

### CODE REVIEW

File: tinlake-ui/components/DebugFlags/context.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react'
+import { debug, flagsConfig } from './config'
+
+export type Key = keyof typeof flagsConfig
+export type Flags = { [key in Key]: any }
+
+export interface Context {
```
Comment (alessioscalici) [inline tinlake-ui/components/DebugFlags/context.ts:7]: If this isn't used anywhere else, should we keep it private?

Or if it is, maybe we could give it a more specific name (e.g. `IDebugFlagsContext`)..

WDYT?

File: tinlake-ui/components/DebugFlags/context.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+let i = 0
+
+export function useDebugFlags() {
+  const [id] = React.useState(() => ++i)
+  const ctx = React.useContext(DebugFlagsContext)
+  const tracked = React.useRef<any>({})
+  const stateRef = React.useRef(ctx.flags)
+  stateRef.current = ctx.flags
+
```
Comment (alessioscalici) [inline tinlake-ui/components/DebugFlags/context.ts:28]: Could you ad a comment explaining the way you used to understand if the flags are used in the page or not?

To a first reader, it's not really obvious from the code

**Review by alessioscalici — CHANGES_REQUESTED** [review]

File: tinlake-ui/components/DebugFlags/context.ts (hunk 1 above)
Comment (onnovisser) [reply tinlake-ui/components/DebugFlags/context.ts:7]: I'll just not export it 👍

File: tinlake-ui/components/DebugFlags/context.ts (hunk 2 above)
Comment (alessioscalici) [reply tinlake-ui/components/DebugFlags/context.ts:33]: thanks!

**Review by alessioscalici — APPROVED** [review]

**Pull request merged**